torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -148,29 +148,47 @@ const std::exception& ProcessGroupNCCL::WorkNCCL::exception() const {
       "isCompleted() and wait() will either succeed or throw");
 }
 
+std::unordered_set<ssize_t> ProcessGroupNCCL::processGroups_;
+ssize_t ProcessGroupNCCL::processGroupCounter_ = -1;
+
 ProcessGroupNCCL::ProcessGroupNCCL(
     const std::shared_ptr<Store>& store,
     int rank,
     int size)
     : ProcessGroup(rank, size), store_(store) {
-  C10D_CUDA_CHECK(cudaGetDeviceCount(&numGPUs_));
-  thcState_ = ::at::globalContext().lazyInitCUDA();
+  // C10D_CUDA_CHECK(cudaGetDeviceCount(&numGPUs_));
+  thcState_ = ::at::globalContext().getTHCState();
+  if (!thcState_) {
+    thcState_ = ::at::globalContext().lazyInitCUDA();
+  }
```

Comment:
Isn't lazy init doing exactly the same -- get or init?

Reply:
This should be removed, my test code